kong/db/dao/certificates.lua

```diff
@@ -53,6 +54,14 @@ function _Certificates:insert(cert, options)
     return nil, err, err_t
   end
 
+  if (not cert.key or cert.key == null) and name_list and #name_list ~= 0 then
+    local err_t = self.errors:schema_violation({
+      snis = "private key is not present, \"snis\" field should be absent",
```

Comment:
One recommendation I'd like to make is to reformulate those errors to be a bit more explicit (instead of "present", "absent", and a couple of semantic/quoting issues). Maybe something like:
```
snis = "cannot be specified since Certificate has no 'key' attribute"
```

(Since `snis` already is the field name in the error object, this is closer to our other schema errors for individual fields. We also use single quotes to refer to other attributes in the error itself, and capitalize entities' names, as per our writing guidelines)

Reply:
not to bikeshed, but `has no 'key' attribute` is also slightly confusing, as the entity itself _has_ the attribute, the instance doesn't (and people sometimes confuse those things). `cannot be specified because the 'key' attribute is not set`, perhaps?

Reply:
Changed to `snis = "cannot be specified because the 'key' attribute is not set"`, I think this sounds the best and aligns with both @thibaultcha and @hishamhm 's suggestions :)